packages/flutter/lib/src/widgets/drag_boundary.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'framework.dart';
+
+/// An abstract class that defines drag boundaries
+///
+/// [T] is the drag object position.
+abstract class DragBoundary<T> {
+  /// Returns whether the specified drag object position is within the boundary.
+  bool isWithinBoundary(T position);
```

Comment:
Is this method called in your future PR?

Reply:
It might not, but this method is very useful when you only need to determine if something is within the boundary, such as canceling the gesture when dragging out of bounds.

Reply:
```suggestion
/// `T` is a data class that defines the shape being dragged. For example, when dragging a rectangle within the boundary, `T` should be a `Rect`. 
```